components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/util/X509CertificateUtil.java

```diff
@@ -1041,18 +1042,14 @@ private static void createAttributeList(Validator validator, Resource resource)
      */
     public static void startTenantFlow(int tenantId) {
 
-        PrivilegedCarbonContext.startTenantFlow();
-        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
-        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
-        carbonContext.setTenantId(tenantId);
-        carbonContext.setTenantDomain(tenantDomain);
-    }
-
-    /**
-     * End tenant flow.
-     */
-    public static void endTenantFlow() {
-
-        PrivilegedCarbonContext.endTenantFlow();
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+            String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+            carbonContext.setTenantId(tenantId);
+            carbonContext.setTenantDomain(tenantDomain);
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
```

Comment:
Is this logic correct? Don't we need to perform some action after setting the tenant id/domain..

This is just start a tenant flow and end it immediately.